server/controllers/userController.js

```diff
@@ -0,0 +1,48 @@
+import uuid from 'uuid';
+import data from '../dummyData/data.json';
+import tokenHelper from './helperFunctions/tokenHelper';
+
+/**
+ * Class userController contains all methods pertaining to users
+ */
+class userController {
+  /**
+   * Creates a new user
+   * @param {Object} req - request parameters
+   * @param {Object} res - response object
+   * @returns {Object} res - returns res object
+   */
+  static signUp(req, res) {
+    const email = data.find(user => user.email === req.body.email);
+    if (email) {
+      return res.status(409).send({
+        status: 409,
+        message: 'email already exists'
+      });
+    }
+    const newUser = {
+      id: uuid.v4(),
+      email: req.body.email.toLowerCase().trim(),
```

Comment:
I think you can write a helper function for this ```toLowerCase().trim()```

something like this 
```
toLowerCaseAndTrim(obj){
   ....
   return newObj;
}
```
You may check here for an efficient way to loop through an object in js https://zellwk.com/blog/looping-through-js-objects/

Reply:
how many lines of code will this save lol

Reply:
I was thinking of  Single responsibility., The S in Solid  :-)

Reply:
Ok i misunderstood @Eazybee  what you wrote from your helper function. Please be more explicit when giving an example so nothing gets lost.

@mosinmiloluwa01  since you have access to the `req.body` you can find a way to loop through the values of this object and then lowercase and trim each value. However, remember not all fields need to be lowerCased and trimmed.